### DISCUSSION TITLE
After upgrade of OSS install on Docker in EC2, permission errors on the /tmp/workspace directory

### DISCUSSION THREAD

**Discussion by techxorcist** [airbytehq/airbyte#36984] [Connector Questions] [answered]
Good day. 

I have upgraded an OSS instance of Airbyte from v0.50.35 to v0.57.2. All pods referred to in the docker-compose start successfully but any source or destination fails on execution of the check() method. In the airbyte-worker logs, there is a file permissions error:

> ERROR StatusConsoleListener FileManager (/tmp/workspace/4de9764c-5e55-4c00-9cf6-f23c54d6db2d/0/logs.log) java.io.IOException: Could not create directory /tmp/workspace/4de9764c-5e55-4c00-9cf6-f23c54d6db2d/0

The /tmp folder is world-writeable in the instance so a file permissions error is puzzling.

**Other points from my investigation:**

- There is discussion of the same issue in [Slack](https://airbytehq.slack.com/archives/C021JANJ6TY/p1711356903827139).

- The [pull request for the draft March 2024 release notes](https://github.com/airbytehq/airbyte/pull/36871) mentions breaking changes to Docker and Helm deployments though the [details in the linked-to information](https://airbyte-docs-git-natalie-release-march-airbyte-growth.vercel.app/deploying-airbyte/on-kubernetes-via-helm#migrate-from-old-chart-to-airbyte-v0520-and-latest-chart-version) seem only concerned with the Helm deployment.

- The [v0.57.2 docker-compose.yaml](https://raw.githubusercontent.com/airbytehq/airbyte-platform/v0.57.2/docker-compose.yaml) has some changes in the volume section for airbyte-worker (and -server and -proxy) since [v0.50.35](https://raw.githubusercontent.com/airbytehq/airbyte-platform/v0.50.35/docker-compose.yaml) but not to the WORKSPACE_ROOT:
![image](https://github.com/airbytehq/airbyte/assets/27961730/18a8df94-2652-48f1-b303-671b0c5a414b)

- The .env the script loads and calls has similar change to a directory location reference but, again, hard to see the relation:
![image](https://github.com/airbytehq/airbyte/assets/27961730/76b0d92c-3fc9-49bc-a8de-1b8cff1cb92e)

Any insight would be very helpful. Thank you.

**Comment by techxorcist** [comment, marked as answer]
Resolved with the explanation on the [March 2024 release notes PR](https://github.com/airbytehq/airbyte/pull/36871) and clarification from @nataliekwong. As mentioned in [this issue](https://github.com/airbytehq/airbyte/issues/36783#issuecomment-2037707927), deleting and recreating the `airbyte_workspace` volume can resolve the issue. It is also possible to resolve the issue by changing the ownership on /tmp with the following commands, run on the host machine:
```
docker exec -u root -it airbyte-worker /bin/bash
chown -R airbyte:airbyte /tmp
```

**Comment by pradosh-suar** [comment]
This issue is happening because of line 121 in DockerProcessFactory.java file which is trying to create a multilevel directory “/tmp/workspace/f54cc18e-ce16-406d-9c5b-b1fe2493cf73/0”  using Files.createDirectory() method. This method is only used to create a subdirectory inside an existing directory not a chain.

Can you please change the code to use Files.createDirectories() method, this will solve the issue. Will be grateful if it get addressed little early, we are blocked.
